src/services/fields.ts

```diff
@@ -179,7 +187,18 @@ export function getLabelTypeFromFrame(labelKey: string, frame: DataFrame, index
   }
 }
 
-export function getFilterTypeFromLabelType(type: LabelType | null, key: string, value: string): VariableFilterType {
+export function getFilterTypeFromLabelType(
+  type: LabelType | null,
+  key: string,
+  value: string,
+  sceneRef: SceneObject
+): VariableFilterType {
```

Comment:
There's something very odd with this function. `type` can be null, or have a value; independent of the type, we check for `getParserForField()`, which supports only `structuredMetadata`, so it can return something else, or nothing, but type can also be null, which goes to the default case.

I think we should refactor this before merging.

Reply:
I'm not sure I fully understood, how does this look? https://github.com/grafana/explore-logs/pull/826/commits/b643f482c1731757ab59979f0a3e0b9c9c841308

Reply:
Better, thanks.